### PR TITLE
Rewritten Sandwich Lady and Mysterious Old Man Randoms

### PR DESCRIPTION
**Pull request by Fryslan**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -68,4 +68,5 @@ fabric.properties
 # virtual machine crash logs, see http://www.java.com/en/download/help/error_hotspot.xml
 hs_err_pid*
 
-*.iml
\ No newline at end of file
+*.iml
+.i€dea
```
Comment (JKetelaar) [inline .gitignore:72]: Seems like something wrong here?
image

File: src/main/java/org/parabot/randoms/pkhonor/MysteriousOldMan.java [hunk 2]
```diff
@@ -12,32 +12,37 @@
  */
 public class MysteriousOldMan implements Random {
 
-    Npc man;
+    private Npc man;
+    private final int ID = 410;
```
Comment (JKetelaar) [inline src/main/java/org/parabot/randoms/pkhonor/MysteriousOldMan.java]: Should be lowercase, as it's private

File: src/main/java/org/parabot/randoms/pkhonor/SandwichLady.java [hunk 3]
```diff
@@ -12,32 +12,37 @@
  */
 public class SandwichLady implements Random {
 
-    Npc lady;
+    private Npc lady;
+    private final int ID = 3117;
```
Comment (JKetelaar) [inline src/main/java/org/parabot/randoms/pkhonor/SandwichLady.java]: Should be lowercase, as it's private

File: src/main/java/org/parabot/randoms/pkhonor/MysteriousOldMan.java [hunk 4]
```diff
@@ -12,32 +12,37 @@
...
                 @Override
                 public boolean isValid() {
-                    return !man.getInteractingCharacter().equals(Players.getMyPlayer());
+                    return man == null || !man.getInteractingCharacter().equals(Players.getMyPlayer());
                 }
             }, 1500);
         }
     }
 
+    private Npc man() {
```
Comment (JKetelaar) [inline src/main/java/org/parabot/randoms/pkhonor/MysteriousOldMan.java]: Rename this to something more likely, like: 

```
getMan()
```

File: src/main/java/org/parabot/randoms/pkhonor/SandwichLady.java [hunk 5]
```diff
@@ -12,32 +12,37 @@
...
                 @Override
                 public boolean isValid() {
-                    return !lady.getInteractingCharacter().equals(Players.getMyPlayer());
+                    return lady == null || !lady.getInteractingCharacter().equals(Players.getMyPlayer());
                 }
             }, 1500);
         }
     }
 
+    private Npc lady() {
```
Comment (JKetelaar) [inline src/main/java/org/parabot/randoms/pkhonor/SandwichLady.java]: Same for this, maybe:

```
getLady()
```

File: src/main/java/org/parabot/randoms/pkhonor/SandwichLady.java [hunk 6]
```diff
@@ -12,32 +12,37 @@
...
     public void execute() {
-        if (lady != null && lady.getDef() != null && lady.getInteractingCharacter().equals(Players.getMyPlayer())) {
-            lady.interact(0);
+        if (this.lady != null) {
+            lady.interact(Npcs.Option.TALK_TO);
             Time.sleep(new SleepCondition() {
                 @Override
                 public boolean isValid() {
-                    return !lady.getInteractingCharacter().equals(Players.getMyPlayer());
+                    return lady == null || !lady.getInteractingCharacter().equals(Players.getMyPlayer());
```
Comment (JKetelaar) [inline src/main/java/org/parabot/randoms/pkhonor/SandwichLady.java]: `lady` doesn't become null out of nowhere, have some check like:

```
lady.getX() > 0
```

File: src/main/java/org/parabot/randoms/pkhonor/MysteriousOldMan.java [hunk 7]
```diff
@@ -12,32 +12,37 @@
...
     public void execute() {
-        if (man != null && man.getDef() != null && man.getInteractingCharacter().equals(Players.getMyPlayer())) {
-            man.interact(0);
+        if (this.man != null) {
+            man.interact(Npcs.Option.TALK_TO);
             Time.sleep(new SleepCondition() {
                 @Override
                 public boolean isValid() {
-                    return !man.getInteractingCharacter().equals(Players.getMyPlayer());
+                    return man == null || !man.getInteractingCharacter().equals(Players.getMyPlayer());
```
Comment (JKetelaar) [inline src/main/java/org/parabot/randoms/pkhonor/MysteriousOldMan.java]: See the same line for `lady`

**Comment by JKetelaar** [comment]
Great work on this @Fryslan, please go through the notes I've added.

**Comment by Fryslan** [comment]
Thanks for review will fix the issues

**Comment by JKetelaar** [comment]
Will be continued in PR #5

**Pull request closed without merging**